### ISSUE TITLE
"'NoneType' object is not subscriptable" error when trying to refresh Steam library in Sources

### ISSUE THREAD

**Issue by RAKtheUndead** [lutris/lutris#5248]
In both 0.5.15 and 0.5.16, I've been trying to refresh my Steam library so that I can sync the playtime between Lutris and Steam, but whenever I've tried pressing the refresh button next to Steam in Sources, I've been receiving an error with the text: 'NoneType' object is not subscriptable

Running Lutris through the CLI provides the following backtrace:

2024-01-17 17:59:23,089: Error while completing task <function BaseService.start_reload.<locals>.do_reload at 0x7f03a2bc7c40>: <class 'TypeError'> 'NoneType' object is not subscriptable
  File "/usr/lib/python3.11/site-packages/lutris/util/jobs.py", line 31, in target
    result = self.function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/lutris/services/base.py", line 145, in do_reload
    self.load()
  File "/usr/lib/python3.11/site-packages/lutris/services/steam.py", line 108, in load
    self.match_games()
  File "/usr/lib/python3.11/site-packages/lutris/services/base.py", line 246, in match_games
    self.match_game(service_games.get(provider_game["slug"]), lutris_game)
  File "/usr/lib/python3.11/site-packages/lutris/services/steam.py", line 115, in match_game
    for game in get_games(filters={"service": self.id, "service_id": service_game["appid"]}):

I'm not sure if this is down to a specific game or games in my Steam library or if this is a wider bug in Lutris, but if it's the latter, is there a way for me to check?

**Comment by danieljohnson2** [comment]
This looks very strange to me, but I've added a check against this None.

What happened here is that there was some additional code to sync steam playtime, but it made GOG crash (and I think other services would have too). I moved it down to the Steam service only. I did not include a check against None though; didn't realize it was needed.

I still can't see why it is, or if this will happen with many Steam libraries. It looks like you have a Steam service game that is known by Lutris.net to actually be some other Steam game... that you don't own. I do not understand how this is possible. If this is a common failure, it could be a problem.

However, for now you can download the master branch with the added check, and then run the 'bin/lutris' script within. No need to install! You should be able to use this to refresh your library.

**Comment by nilfilter** [comment]
Same here on Opensuse TW:
````
$ lutris 
2024-01-24 01:40:02,425: Starting Lutris 0.5.16
2024-01-24 01:40:02,426: Running AMD Mesa driver 23.2.1 on AMD Radeon RX 6600 XT (navi23, LLVM 17.0.6, DRM 3.54, 6.6.11-1-default) (0x73ff)
2024-01-24 01:40:02,426: GPU: xxx (amdgpu drivers)
2024-01-24 01:40:11,758: Error while completing task <function BaseService.start_reload.<locals>.do_reload at 0x7f642c475300>: <class 'TypeError'> 'NoneType' object is not subscriptable
  File "/usr/lib/python3.11/site-packages/lutris/util/jobs.py", line 31, in target
    result = self.function(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/lutris/services/base.py", line 145, in do_reload
    self.load()
  File "/usr/lib/python3.11/site-packages/lutris/services/steam.py", line 108, in load
    self.match_games()
  File "/usr/lib/python3.11/site-packages/lutris/services/base.py", line 252, in match_games
    self.match_game(service_games.get(provider_game["slug"]), lutris_game)
  File "/usr/lib/python3.11/site-packages/lutris/services/steam.py", line 115, in match_game
    for game in get_games(filters={"service": self.id, "service_id": service_game["appid"]}):
````

> However, for now you can download the master branch with the added check, and then run the 'bin/lutris' script within. No need to install! You should be able to use this to refresh your library.

That worked well, thanks. While the error message still appears, I was able to add a newly released and installed Steam game to my library (Dominions 6 - Rise of the Pantokrator).

**Comment by danieljohnson2** [comment]
Is the error unchanged? If somehow this can still happen in master, it should be line *116* now, and your log says 115.

It really should no longer happen with the latest code, though.

**Comment by nilfilter** [comment]
> It really should no longer happen with the latest code, though.

Sorry, I didn't make myself clear:
It doesn't appear anymore when using the executable from the cloned repo, only when switching back to the Lutris version installed from the distro's official repo.

**Comment by danieljohnson2** [comment]
Might be worth the experiment. But I am not sure how to fix this, even if we know the issue is missing 'lutris_slug' entries, though.

You may be right that the source of the original bug was having versions of the game for different platforms, but if that were all it was I'd expect the *correct* platform to be there, and for the playtime transfer to work because of it.

**Comment by RAKtheUndead** [comment]
Tried another game on Steam, _Erra: Exordium_ which appears to have a missing `lutris_slug` field in pga.db using the master branch executable. As I had suspected, the playtime for this game wasn't registered against the entry in Lutris, although I'm still not sure on whether that's the actual cause of the original issue. 

Most of the other entries in my library with a missing `lutris_slug` entry are either separate Steam entries for the same game/expansion (_Sid Meier's Civilization IV_ and two of its expansions appear to have separate executables for the Mac versions) or expansions for a base game which are for some reason treated separately by Steam despite requiring a base game.
